sdk/servicebus/service-bus/src/diagnostics/instrumentServiceBusMessage.ts

```diff
@@ -54,3 +61,92 @@ export function extractSpanContextFromServiceBusMessage(
   const diagnosticId = message.properties[TRACEPARENT_PROPERTY] as string;
   return extractSpanContextFromTraceParentHeader(diagnosticId);
 }
+
+/**
+ * Provides an iterable over messages, whether it is a single message or multiple
+ * messages.
+ *
+ * @param receivedMessages A single message or a set of messages
+ * @internal
+ * @ignore
+ */
+function* getReceivedMessages(
+  receivedMessages: ServiceBusReceivedMessage | ServiceBusReceivedMessage[]
+): Iterable<ServiceBusReceivedMessage> {
+  if (!Array.isArray(receivedMessages)) {
+    yield receivedMessages;
+  } else {
+    for (const message of receivedMessages) {
+      yield message;
+    }
+  }
+}
+
+/**
+ * A span that encompasses the period when the message has been received and
+ * is being processed.
+ *
+ * NOTE: The amount of time the user would be considered processing the message is
+ * not always clear - in that case the span will have a very short lifetime
+ * since we'll start the span when we receive the message and end it when we
+ * give the message to the user.
+ *
+ * @internal
+ * @ignore
+ */
+export function createProcessingSpan(
+  receivedMessages: ServiceBusReceivedMessage | ServiceBusReceivedMessage[],
+  // NOTE: the connectionConfig also has an entityPath property but that only
+  // represents the optional entityPath in their connection string which is NOT
+  // what we want for tracing.
+  receiver: Pick<ServiceBusReceiver<any>, "entityPath">,
+  connectionConfig: Pick<ConnectionContext["config"], "host">,
+  options?: OperationOptionsBase
+): Span {
+  const links: Link[] = [];
+
+  for (const receivedMessage of getReceivedMessages(receivedMessages)) {
+    const spanContext = extractSpanContextFromServiceBusMessage(receivedEvent);
+
+    if (spanContext == null) {
+      continue;
+    }
+
+    links.push({
+      context: spanContext,
+      attributes: {
+        enqueuedTime: receivedEvent.enqueuedTimeUtc?.getTime()
```

Comment:
There are more spots of `receivedEvent`. 🙂

Reply:
Dang it. I'll do some search and replacing. :)